### ISSUE TITLE
about datasize

### ISSUE THREAD

**Issue by qgking** [sinhasam/vaal#12]
Hi,
Thanks for your great work. I have two questions:
1. the batch size of labeled/unlabelled dataloader is the same, while the total number of labeled/unlabeled data set is different. That is, when one type of data is all trained, the other one still has data left. Am I right?
2. In your code, part of unlabelled data is used to train VAE and Discriminator, the sampler still select uncertain data from all the unlabelled data set (including those parts for training VAE and Discriminator). Most likely the sampler will choose the trained unlabelled data. 

Thank you for your time.

**Comment by sinhasam** [comment]
Thanks for your interest!

1. I’m not sure what you mean, but hopefully I can explain this well. The unlabelled data and labelled data have no intersection (no points exist in both, since if it has a label for it then it will be in the labelled pool, otherwise it’ll be in the unlabelled pool). In practice, you can implement that using the indices  

2. It’s very important for us to be able to use the unlabelled data, since we assume that unlabelled data is available during training, to train the VAE+the discriminator. 

Feel free to follow up via email where I’m usually more active.
